### PR TITLE
[WIP] 151 thread wrappers

### PR DESCRIPTION
**Pull request by ArjunBhuptani**
See connected issue

### CODE REVIEW

**Comment by ArjunBhuptani** [comment]
Need to test these directly, what's the best way to do this? It's a little tough for threads because I'll need to prep thread initial states and generate proofs. Will try writing a basic tester for this later today

File: src/Connext.ts [hunk 1]
```diff
@@ -410,6 +410,214 @@ export class ChannelManager implements IChannelManager {
     return new Web3TxWrapper(this.address, 'userAuthorizedUpdate', call.send(sendArgs))
   }
 
+  async startExit(state: ChannelState) {
```
Comment (LayneHaber) [inline src/Connext.ts]: are these functions added to the `IChannelManager` interface?

File: src/Connext.ts [hunk 2]
```diff
@@ -410,6 +410,214 @@ export class ChannelManager implements IChannelManager {
...
+    const sendArgs = {
+      from: state.user,
+      value: 0,
+    } as any
+    const gasEstimate = await call.estimateGas(sendArgs)
+    sendArgs.gas = toBN(gasEstimate * this.gasMultiple)
+    return new Web3TxWrapper(this.address, 'startExitWithUpdate', call.send(sendArgs))
+  }
+
+  async emptyChannelWithChallenge(state: ChannelState) {
```
Comment (LayneHaber) [inline src/Connext.ts]: Did you want to make a `normalize` function for these arguments, or are you going to keep that in the contract tests?

File: src/Connext.ts (hunk 1 above)
Comment (LayneHaber) [reply src/Connext.ts]: Also, you should specify a return type here for clarity

**Comment by LayneHaber** [comment]
What do you mean when you say "test these directly"? unit test them? test them on the contract?

**Pull request merged**